lib/ansible/modules/user.py

```diff
@@ -3114,7 +3114,12 @@ class BusyBox(User):
         - remove_user()
         - modify_user()
     """
+    def _validate_shell(self):
+        with open("/etc/shells", "rb") as f:
+            shells = [line.decode().strip() for line in f.readlines() if not line.startswith(b"#")]
```

Comment:
We should probably filter blank lines too (the change assumes file open in text mode which is how I tested it):

```suggestion
            shells = [l for line in f.readlines() if (l := line.strip()) and not l.startswith("#")]
```

Reply:
Sure thing!

Reply:
I went with a nested comprehension using a generator just to keep the module more portable.